appserver/tests/payara-samples/samples-programmatic/pom.xml

```diff
@@ -29,4 +29,26 @@
             <artifactId>junit</artifactId>
         </dependency>
     </dependencies>
+
+    <profiles>
+        <profile>
+            <id>payara-server-remote</id>
+            <properties>
+                <exclude.tests>**/programatic/**.java</exclude.tests>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <exclude>${exclude.tests}</exclude>
```

Comment:
Exclude explicitly rather than using a property? Or possibly exclude the module from the list of included modules if that's how the project is set up

Reply:
We want the tests to run when not using remote profile

Reply:
See above comment

Reply:
I could exclude the module instead, but I dont think its neccessery